src/Tools.pm

```diff
@@ -484,42 +524,26 @@ returns 1 if yes, 0 if no
 
 =cut
 
-sub IsDMRaidSlave {
-    my $disk = shift;
-
-    unless ( DMRaidAvailable() and -e $dmsetup) {
-        return 0;
-    }
+sub IsDMRaidSlave
+{
+  my $self = $lib_ref;
+  my $dev = shift;
 
-    my $majmin_disk = `stat -c "%t:%T" $disk`;
-    chomp($majmin_disk);
-    my @dmparts = ();
+  my $r = $self->{cache}{IsDMRaidSlave}{$dev};
 
+  if(defined $r) {
+    $self->debug("IsDMRaidSlave($dev) (cached) = $r");
 
-    my @dm_devs = qx{$dmsetup info -c --noheadings -o name | grep -v part};
-    chomp @dm_devs;
+    return $r;
+  }
 
-    if ($dm_devs[0] !~ /No devices found/) {
-        foreach my $dmdisk (@dm_devs) {
-            my @tables = qx{$dmsetup table '$dmdisk'};
-            chomp @tables;
+  my @d = glob Bootloader::Path::Prefix("/sys/block/$dev/holders/*");
```

Comment:
please use more descriptive variable name. It is quite hard to guess what `d` means here.